tests/test_models.py

```diff
@@ -1197,3 +1198,100 @@ def test_apcorr_none_array():
     for name in ("ap_corrections", "ee_fractions", "ee_radii", "sky_background_rin", "sky_background_rout"):
         setattr(m.data.GRISM, name, None)
     assert m.validate() is None
+
+
+def test_make_wfi_wcs():
+    """
+    Check create of WfiWcsModel.
+    """
+    m = utils.mk_datamodel(datamodels.WfiWcsModel)
+
+    assert m.validate() is None
+
+
+@pytest.mark.parametrize(
+    "mk_model",
+    [lambda: datamodels.ImageModel(utils.mk_level2_image(shape=(8, 8)))],
+)
+def test_wfi_wcs_from_wcsmodel(mk_model):
+    model = mk_model()
+
+    # Give the model's WCS a bounding box.
+    model.meta.wcs.bounding_box = ((-0.5, 4087.5), (-0.5, 4087.5))
+
+    # Give the model some alignment results
+    model.meta["wcs_fit_results"] = {
+        "<rot>": 1.2078100852299566e-05,
+        "<scale>": 1.0,
+        "center": [-3.090428960153321, -18.122328864329525],
+        "fitgeom": "rshift",
+        "mae": 0.0017789920274088183,
+        "matrix": [[0.9999999999999778, 2.108026272605592e-07], [-2.108026272605592e-07, 0.9999999999999778]],
+        "nmatches": 109,
+        "proper": True,
+        "proper_rot": 1.2078100852299566e-05,
+        "rmse": 0.0022859902707182554,
+        "rot": [1.2078100852299566e-05, 1.2078100852299566e-05],
+        "scale": [1.0, 1.0],
+        "shift": [0.00017039070698617517, -0.00023752675967125825],
+        "skew": 0.0,
+        "status": "SUCCESS",
+    }
+
+    wfi_wcs = datamodels.WfiWcsModel.from_model_with_wcs(model)
+
+    # Test for equality of attributes
+    for key in wfi_wcs:
+        if not hasattr(model, key):
+            continue
+
+        wfi_wcs_value = getattr(wfi_wcs, key)
+        model_value = getattr(model, key)
+        if isinstance(wfi_wcs_value, np.ndarray):
+            assert_array_equal(wfi_wcs_value, model_value.astype(wfi_wcs_value.dtype))
+
+        elif key == "meta":
+            wfi_wcs_meta = wfi_wcs_value.to_flat_dict(include_arrays=False, recursive=True)
+            model_meta = model_value.to_flat_dict(include_arrays=False, recursive=True)
+            for meta_key in wfi_wcs_meta:
+                if meta_key == "model_type":
+                    wfi_wcs_value[meta_key] = wfi_wcs.__class__.__name__
+                    model_value[meta_key] = model.__class__.__name__
+                    continue
+                elif meta_key == "cal_step":
+                    continue
+                if meta_key in model_meta:
+                    assert wfi_wcs_meta[meta_key] == model_meta[meta_key]
+
+        elif isinstance(wfi_wcs_value, stnode.DNode):
+            assert_node_equal(wfi_wcs_value, model_value)
+
+        else:
+            raise ValueError(f"Unexpected type {type(wfi_wcs_value)}, {key}")  # pragma: no cover
```

Comment:
```suggestion
    for key in wfi_wcs.meta:
        wfi_wcs_value = getattr(wfi_wcs.meta, key)
        model_value = getattr(model.meta, key)
        assert_node_equal(wfi_wcs_value, model_value)
```
Since this doesn't test `wcs_l1` or `wcs_l2` (they don't exist in `model`) this block is only testing `meta` (see the lack of coverage of the test). It can be simplified to the above suggestion.

Reply:
It does have a wcs. Breaking at #1218 gives:
```
(Pdb) p model.meta.wcs
<WCS(output_frame=icrs, input_frame=detector, forward_transform=Model: CompoundModel
Inputs: ('x0', 'x1')
Outputs: ('alpha_C', 'delta_C')
Model set size: 1
Expression: [0] & [1] | [2] & [3] | [4] | [5]
Components:
    [0]: <Shift(offset=-500.)>

    [1]: <Shift(offset=-500.)>

    [2]: <Scale(factor=0.00002778)>

    [3]: <Scale(factor=0.00002778)>

    [4]: <Pix2Sky_Gnomonic()>

    [5]: <RotateNative2Celestial(lon=30., lat=45., lon_pole=180.)>
Parameters:
    offset_0 offset_1        factor_2              factor_3             lon_5        lat_5 lon_pole_5
    -------- -------- --------------------- --------------------- ------------------ ----- ----------
      -500.0   -500.0 2.777777777777778e-05 2.777777777777778e-05 29.999999999999996  45.0      180.0)>
```

and after line 1240 which is
```
    wfi_wcs = datamodels.WfiWcsModel.from_model_with_wcs(model)
```
gives `wfi_wcs`:
```
(Pdb) p wfi_wcs.wcs_l1
<WCS(output_frame=icrs, input_frame=detector, forward_transform=Model: CompoundModel
Inputs: ('x0', 'x1')
Outputs: ('alpha_C', 'delta_C')
Model set size: 1
Expression: [0] & [1] | [2] & [3] | [4] & [5] | [6] | [7]
Components:
    [0]: <Shift(offset=-4.)>

    [1]: <Shift(offset=-4.)>

    [2]: <Shift(offset=-500.)>

    [3]: <Shift(offset=-500.)>

    [4]: <Scale(factor=0.00002778)>

    [5]: <Scale(factor=0.00002778)>

    [6]: <Pix2Sky_Gnomonic()>

    [7]: <RotateNative2Celestial(lon=30., lat=45., lon_pole=180.)>
Parameters:
    offset_0 offset_1 offset_2 offset_3        factor_4              factor_5             lon_7        lat_7 lon_pole_7
    -------- -------- -------- -------- --------------------- --------------------- ------------------ ----- ----------
        -4.0     -4.0   -500.0   -500.0 2.777777777777778e-05 2.777777777777778e-05 29.999999999999996  45.0      180.0)>
```

Reply:
Oh, but I see what you are getting at. Will include the test of the wcs's also.